Source/WebCore/rendering/RenderObject.cpp

```diff
@@ -2728,7 +2766,12 @@ auto RenderObject::collectSelectionGeometriesInternal(const SimpleRange& range)
                 lineBottom = std::min(lastLineTop, lineBottom);
         } else {
             adjustLineHeightOfSelectionGeometries(geometries, i, lineNumber, lineTop, lineBottom - lineTop);
-            if (!hasFlippedWritingMode) {
+            if (geometries[i].separateBoundary()) {
+                lastLineTop = std::numeric_limits<int>::max();
+                lastLineBottom = std::numeric_limits<int>::min();
+                lineTop = currentRectTop;
+                lineBottom = currentRectBottom;
```

Comment:
Would we still want to coalesce selection geometries in the case where there's two adjacent selected text nodes, that are _both_ within the same out-of-flow positioned renderer?

Reply:
Yes, we will coalesce the geometries if both are in the same positioned renderer.